chunky/src/java/se/llbit/chunky/renderer/RenderWorkerPool.java

```diff
@@ -149,4 +170,42 @@ public void interrupt() {
       worker.interrupt();
     }
   }
+
+  /**
+   * The future to a job. It may be waited on to finish with {@code awaitFinish}.
+   */
+  public static class RenderJobFuture {
+    private volatile boolean done = false;
+    protected final RenderJob task;
+
+    protected RenderJobFuture(RenderJob task) {
+      this.task = task;
+    }
+
+    protected synchronized void finished() {
+      this.done = true;
```

Comment:
is this actually atomic and thread-safe?

Reply:
Yes, the method is synchronized. It is also unconditionally setting a boolean to true which shouldn't have any issues.